pkg/controllers/management/auth/user_test.go

```diff
@@ -0,0 +1,720 @@
+				)
+			},
+			expectedError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.mockSetup()
+
+			err := ul.deleteAllCRTB(tt.inputCRTB)
+
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_deleteAllPRTB(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	prtbMock := NewMockProjectRoleTemplateBindingInterface(ctrl)
+
+	ul := &userLifecycle{
+		prtb: prtbMock,
+	}
+
+	tests := []struct {
+		name          string
+		inputPRTB     []*v3.ProjectRoleTemplateBinding
+		mockSetup     func()
+		expectedError bool
+	}{
+		{
+			name: "remove namespaced prtb",
+			inputPRTB: []*v3.ProjectRoleTemplateBinding{
+				&v3.ProjectRoleTemplateBinding{
+					UserName: "testuser",
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testprtb",
+						Namespace: "testprtbns",
+					},
+				},
+			},
+			mockSetup: func() {
+				prtbMock.EXPECT().DeleteNamespaced("testprtbns", "testprtb", &metav1.DeleteOptions{}).Return(nil)
+			},
+			expectedError: false,
+		},
+		{
+			name: "remove all prtb",
+			inputPRTB: []*v3.ProjectRoleTemplateBinding{
+				&v3.ProjectRoleTemplateBinding{
+					UserName: "testuser",
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testprtb",
+						Namespace: "testprtbns",
+					},
+				},
+				&v3.ProjectRoleTemplateBinding{
+					UserName: "testuser2",
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "testprtb2",
+					},
+				},
+			},
+			mockSetup: func() {
+				gomock.InOrder(
+					prtbMock.EXPECT().DeleteNamespaced("testprtbns", "testprtb", &metav1.DeleteOptions{}).Return(nil),
+					prtbMock.EXPECT().Delete("testprtb2", &metav1.DeleteOptions{}).Return(nil),
+				)
+			},
+			expectedError: false,
+		},
+		{
+			name: "error deleting namespaced prtb",
+			inputPRTB: []*v3.ProjectRoleTemplateBinding{
+				&v3.ProjectRoleTemplateBinding{
+					UserName: "testuser",
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testprtb",
+						Namespace: "testprtbns",
+					},
+				},
+			},
+			mockSetup: func() {
+				prtbMock.EXPECT().DeleteNamespaced("testprtbns", "testprtb", &metav1.DeleteOptions{}).Return(fmt.Errorf("error"))
+			},
+			expectedError: true,
+		},
+		{
+			name: "error deleting prtb",
+			inputPRTB: []*v3.ProjectRoleTemplateBinding{
+				&v3.ProjectRoleTemplateBinding{
+					UserName: "testuser",
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "testprtb",
+					},
+				},
+			},
+			mockSetup: func() {
+				prtbMock.EXPECT().Delete("testprtb", &metav1.DeleteOptions{}).Return(fmt.Errorf("error"))
+			},
+			expectedError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.mockSetup()
+
+			err := ul.deleteAllPRTB(tt.inputPRTB)
+
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_deleteUserNamespace(t *testing.T) {
```

Comment:
For completeness, could you add a test where it getting the namespace returns a "NotFound" error? And a test case for when the namespace is in `NamespaceTerminating` phase?